src/Aspire.Hosting/Dcp/ResourceLogSource.cs

```diff
@@ -24,25 +24,42 @@ public async IAsyncEnumerator<LogEntryList> GetAsyncEnumerator(CancellationToken
             throw new ArgumentException("Cancellation token must be cancellable in order to prevent leaking resources.", nameof(cancellationToken));
         }
 
-        var timestamps = resource is Container; // Timestamps are available only for Containers as of Aspire P5.
-        var stdoutStream = await kubernetesService.GetLogStreamAsync(resource, Logs.StreamTypeStdOut, follow: true, timestamps: timestamps, cancellationToken).ConfigureAwait(false);
-        var stderrStream = await kubernetesService.GetLogStreamAsync(resource, Logs.StreamTypeStdErr, follow: true, timestamps: timestamps, cancellationToken).ConfigureAwait(false);
-
         var channel = Channel.CreateUnbounded<LogEntry>(new UnboundedChannelOptions
         {
             AllowSynchronousContinuations = false,
             SingleReader = true,
             SingleWriter = false
         });
 
+        var streamTasks = new List<Task>();
+
+        var timestamps = resource is Container; // Timestamps are available only for Containers as of Aspire P5.
+
+        if (resource is Container)
```

Comment:
nit:
```suggestion
        if (timestamps)
```

Reply:
Will look at this as a follow up. You would be surprised how much I agonized over that.